### ISSUE TITLE
Issues with group sharing

### ISSUE THREAD

**Issue by ronm123** [thexerteproject/xerteonlinetoolkits#1071]
This is in response to a forum post https://xerte.org.uk/index.php/en/forum/bugs-and-issues/2796-group-sharing-permissions#7875 where it was reported that where a user is part of a group and where the shared project they have been given access to is shared with co-author access the shared users cannot duplicate the project.

I tested this and it seems the duplicated project is created in USER-FILES (complete with all files) but doesn't appear in that user's workspace either inside or outside the group. BTW the duplicate by the shared user doesn't appear in the creator's workspace either but I guess that's correct and to be expected. Also if the creator duplicates the project from the group folder the copy appears outside of the group in the creator's workspace so again I guess that's correct and to share that copy with the group they would need to share that like they would a new project.

While I was testing this I also checked if a group member with co-author access could give a copy of a group project to another user. That appears to work in terms of the confirmation message is shown when giving but the folder created in USER-FILES doesn't contain any xml and although the project appears in the recipients workspace it is not editable and obviously because it doesn't contain any xml. I tested this with xot projects so not 100% sure if the same thing happens with Bootstrap projects. 

So at least two issues here:

- [ ] Group members with co-author access appear to be able to duplicate shared projects (and folder is created) but don't then see them in their workspace.
- [ ] If group members with co-author access give a copy of a project to another user the folder is created but without xml and shows up in the recipients workspace but isn't editable.

@torinfo is this something you are aware of?

**Comment by ronm123** [comment]
BTW after having tested this in a workspace where I have about 6 folders and then lots of individual projects in the root I notice that the group folder appears at the very bottom of the list and therefore can be easily missed. Using the sorting options doesn't appear to affect the position of the group icon/folder. Not sure if anything can be done about this e.g. ensure that the group folders appear below the user's folders?

**Comment by torinfo** [comment]
@ronm123 No, I was not aware of this, so thanks for confirming the issue raised on the Forum (I did see that one, but have not have time to check)

The fact that the group folder is shown last is by design, but I do agree it would make more sense to show them after the user's normal folders. I'll see if we can improve on that.

**Comment by FayCross** [comment]
@torinfo @ronm123 this seems to be the same issue that I emailed about earlier in the week, so it looks like it's not something that's newly broken but has always been a problem.

**Comment by ronm123** [comment]
@FayCross @torinfo does sound like the same issue and any solution should ideally include a check of if/where this has already happened in an install and at least report that if not fix it? Potentially there could be numerous instances where users have tried this and it hasn't worked and perhaps left folders and db entries that don't work?
